ui/app/src/components/Dialogs/PathBrowserDialog.tsx

```diff
@@ -0,0 +1,254 @@
+        setInvalidPath(false);
+      } else {
+        const allPaths = getAllPaths(currentPath, rootPath).filter(
+          path => !nodesLookup[path] || !nodesLookup[path].fetched
+        );
+        const requests: Observable<AjaxResponse>[] = [];
+        allPaths.forEach((nextPath) => {
+          requests.push(
+            get(
+              `/studio/api/1/services/api/1/content/get-items-tree.json?site=${site}&path=${nextPath}&depth=1&order=default`
+            )
+          );
+        });
+
+        forkJoin(requests).subscribe((responses) => {
+          let rootNode;
+          responses.forEach(({ response: { item } }, i) => {
+            let parent;
+
+            if (i === requests.length - 1) {
+              setInvalidPath(item.deleted);
+            }
+
+            if (item.deleted) {
+              return;
+            }
+
+            if (!nodesLookup['root']) {
+              parent = {
+                id: item.path,
+                name: item.name,
+                fetched: true,
+                children: legacyItemsToTreeNodes(item.children)
+              };
+              rootNode = parent;
+              nodesLookup[item.path] = parent;
+              nodesLookup['root'] = parent;
+            } else {
+              rootNode = nodesLookup['root'];
+              parent = nodesLookup[item.path];
+              parent.fetched = true;
+              parent.children = legacyItemsToTreeNodes(item.children);
+            }
+
+            parent.children.forEach((child) => {
+              nodesLookup[child.id] = child;
+            });
+
+          });
+          setTreeNodes({ ...rootNode });
+        });
+      }
+    }
+
+  }, [currentPath, rootPath, site]);
+
+  const resource = useLogicResource<TreeNode, TreeNode>(treeNodes, {
+    shouldResolve: (treeNodes) => Boolean(treeNodes),
+    shouldReject: (treeNodes) => false,
+    shouldRenew: (treeNodes, resource) => treeNodes === null && resource.complete,
+    resultSelector: (treeNodes) => treeNodes,
+    errorSelector: (treeNodes) => null
+  });
+
+  const onCreateFolder = () => {
+    setCreateFolder(true);
+  };
+
+  const onCloseCreateFolder = () => {
+    setCreateFolder(false);
+  };
+
+  const onFolderCreated = (path: string, name: string) => {
+    setCreateFolder(false);
+    let id = `${path}/${name}`;
+    nodesLookupRef.current[currentPath].children.push({
+      id,
+      name: name,
+      children: []
+    });
+    nodesLookupRef.current[id] = {
+      id,
+      name: name,
+      children: [],
+      fetched: true
+    };
+    setCurrentPath(id);
+    setTreeNodes({ ...treeNodes });
+  };
+
+  const onNodeToggle = (event: React.ChangeEvent<{}>, nodeIds: string[]) => {
+    setExpanded(nodeIds);
+  };
+
+  const onNodeSelected = (event: React.ChangeEvent<{}>, nodeId: string) => {
+    setCurrentPath(nodeId);
+  };
+
+  const onPathChanged = (path: string) => {
+    setCurrentPath(path);
+    setExpanded(getAllPaths(path, rootPath));
+  };
```

Comment:
While verifying a path, the accept button should be disabled.

When there's a manual path entry on the input, if the connection is slow enough, there's a moment when you could accept a path that does not exist.